torch/utils/collect_env.py

```diff
@@ -225,6 +234,131 @@ def get_nvidia_smi():
     return smi
 
 
+def get_pkg_version(run_lambda, pkg):
```

Comment:
The package version could be simple use by `sycl-ls` after sourcing oneAPI, that would have output like below, maybe we could use this for simplicity?

```
+XPU:
+XPU Device Driver:
+[opencl:gpu][opencl:3] Intel(R) OpenCL Graphics, Intel(R) Data Center GPU Max 1550 OpenCL 3.0 NEO  [24.22.xxxxx.yy]
+[opencl:gpu][opencl:4] Intel(R) OpenCL Graphics, Intel(R) Data Center GPU Max 1550 OpenCL 3.0 NEO  [24.22.xxxxx.yy]
+[level_zero:gpu][level_zero:0] Intel(R) Level-Zero, Intel(R) Data Center GPU Max 1550 1.3 [1.3.xxxxx]
+[level_zero:gpu][level_zero:1] Intel(R) Level-Zero, Intel(R) Data Center GPU Max 1550 1.3 [1.3.xxxxx]
+Compiler version:
+Intel(R) oneAPI DPC++/C++ Compiler 2024.2.1 (2024.2.1.20240703)

```

Reply:
`sycl-ls` is a separate command that requires users to install separately. Not planning to request users to install more packages.